core/src/main/java/org/infinispan/configuration/cache/InvocationBatchingConfigurationBuilder.java

```diff
@@ -50,6 +50,8 @@ private void enableTransactions() {
             getBuilder().transaction().attributes.attribute(TRANSACTION_MODE);
       if (!transactionModeAttribute.isModified()) {
          getBuilder().transaction().transactionMode(TRANSACTIONAL);
+         getBuilder().transaction().useSynchronization(true);
+         getBuilder().transaction().recovery().enabled(false);
```

Comment:
For reference, when deserializing the configuration:

https://github.com/infinispan/infinispan/blob/ceae0e1d85aeb6cb797b881a457a288ae68afcc5/core/src/main/java/org/infinispan/configuration/parsing/CacheParser.java#L566-L570